.circleci/config.yml

```diff
@@ -272,6 +276,71 @@ jobs:
             docker login --username "${DOCKER_USER}" --password "${DOCKER_PASSWORD}"
             ./gradlew --no-daemon --parallel "-Pbranch=${CIRCLE_BRANCH}" dockerUpload
 
+  extractAPISpec:
+    executor: medium_executor
+    steps:
+      - prepare
+      - attach_workspace:
+          at: ~/project
+      - run:
+          name: Build runnable Teku binaries
+          command: |
+            ./gradlew --no-daemon --parallel installDist
+      - run:
+          name: Extract OpenAPI JSON
+          command: |
+            build/install/teku/bin/teku --network=mainnet --Xinterop-enabled=true --rest-api-docs-enabled=true &
```

Comment:
it might be worth redirecting the output of this command to a file and including that file in the build artifacts. This will make it so that you can inspect them if Teku falls over during the build.

Something like `build/install/teku/bin/teku --network=mainnet --Xinterop-enabled=true --rest-api-docs-enabled=true 2>&1 > teku_output.log &`

and then add `teku_output.log` to your `store_artifacts` list.

Reply:
Also, it might be worth looking into a gradle plugin that can do this at compile time rather than having to actually start a full instance of teku and grab it via the HTTP interface. I _think_ this plugin might do that for you: https://plugins.gradle.org/plugin/org.openapi.generator

Reply:
I already searched for this kind of plugin, but I did not find any that could fit our architecture (gradle + javalin annotations). The openapi generator one is not doing that, it's generating code and doc from spec, but it doesn't generate spec. The closest one would be https://github.com/openapi-tools/swagger-maven-plugin but it's not matching our tools for what I read.

Reply:
made the change on logging output